### PR TITLE
Few Small Updates

### PR DESCRIPTION
**Pull request by FKabir**
- Adding submit button for users when entering their name
- Adding menu option to reset user name
- Adding functionality to determine starting player on game start and pass that back to the client

Most of the updates in `room.js` in the root is just prettier.

### CODE REVIEW

File: room.js [hunk 1]
```diff
@@ -1,160 +1,179 @@
...
+    socket.emit("character", character);
+  }
 
-        this.io.to(this.setup.name).emit('startGame')
-    }
-
-    _assignCharacter(socket, character) {
-        socket.emit('character', character)
-    }
+  _determineStartingPlayer(players) {
```
Comment (itsjamie) [inline room.js]:
```suggestion
  _determineStartingPlayer(players, num) {
```

File: room.js [hunk 2]
```diff
@@ -1,160 +1,179 @@
...
 
-    async startGame() {
-        const num = await generateNumber(0, 1000)
-        const shuffledCards = shuffle(this.setup.characters, () => num / 1000)
-        for (let i = 0; i < shuffledCards.length; i++) {
-            this._assignCharacter(this.players[i].socket, shuffledCards[i])
-        }
+    this.io
+      .to(this.setup.name)
+      .emit("startingPlayer", _determineStartingPlayer(this.players));
```
Comment (itsjamie) [inline room.js]:
```suggestion
      .emit("startingPlayer", _determineStartingPlayer(this.players, num));
```

**Comment by FKabir** [comment]
Thanks for catching that!

**Comment by itsjamie** [comment]
No, thank you!

**Pull request merged**